### PR TITLE
Add password to delete account form

### PR DESCRIPTION
**Pull request by andrebola**
issue #1025

Added password field to account delete form. Also changed how the form is validated and the error messages are displayed.

### CODE REVIEW

**Comment by bdejong** [comment]
YEAH! 👍

File: accounts/forms.py [hunk 1]
```diff
@@ -312,8 +312,15 @@ class DeleteUserForm(forms.Form):
     encrypted_link = forms.CharField(widget=forms.HiddenInput())
     delete_sounds = forms.ChoiceField(choices = DELETE_CHOICES,
             widget=forms.RadioSelect())
+    password = forms.CharField(label=_("Your password"), widget=forms.PasswordInput)
```
Comment (bdejong) [inline accounts/forms.py]: `Confirm your password`
perhaps?

File: accounts/forms.py [hunk 2]
```diff
@@ -312,8 +312,15 @@ class DeleteUserForm(forms.Form):
...
 
-    def clean_encrypted_link(self):
+    def clean_password(self):
+        user = User.objects.get(id=self.user_id)
+        if not user.check_password(self.cleaned_data["password"]):
+            raise forms.ValidationError(_("Incorrect password."))
+        return self.cleaned_data['password']
```
Comment (bdejong) [inline accounts/forms.py]: I guess you should return `None` or `""` here not to have the password be played back to the user? If the user messed up the password is still going to be repeated to the user from the backend... That's a bit messy.

File: accounts/tests.py [hunk 3]
```diff
@@ -668,6 +668,17 @@ def test_user_delete_remove_sounds(self, delete_sound_solr):
...
+        # This should set user's attribute active to false and anonymize it
+        user = self.create_user_and_content(is_index_dirty=False)
+        a = self.client.login(username=user.username, password='testpass')
+        form = DeleteUserForm(user_id=user.id)
+        encr_link = form.initial['encrypted_link']
+        resp = self.client.post(reverse('accounts-delete'),
+                {'encrypted_link': encr_link, 'password': 'testpass', 'delete_sounds': 'delete_sounds'})
+
+        self.assertEqual(User.objects.get(id=user.id).profile.is_deleted_user, True)
+
```
Comment (bdejong) [inline accounts/tests.py]: Should probably add a failing test too, especially one where the password is mistyped...

File: templates/accounts/delete.html [hunk 4]
```diff
@@ -15,15 +15,15 @@
...
     <form method="post">
-        {% if error_message %}
-        <ul class="errorlist nonfield">
-            <li>{{ error_message }}</li>
-        </ul>
-        {% endif %}
       {% csrf_token %}
+      {{ delete_form.non_field_errors }}
       {{delete_form.encrypted_link}}
       {{delete_form.delete_sounds}}
```
Comment (bdejong) [inline templates/accounts/delete.html]: This is weird... Why is {{delete_form}} itself not in here?

File: accounts/tests.py (hunk 3 above)
Comment (ffont) [reply accounts/tests.py]: Yes, this test should be added.

**Pull request merged**